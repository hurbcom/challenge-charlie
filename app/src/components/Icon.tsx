import React from 'react';
import { styled } from 'utils';

interface Props {
  icon: string;
  size?: string;
  color?: string;
}

interface IconProps {
  size?: string;
  'data-icon': string;
  color?: string;
}

const IconContainer = styled.i<IconProps>`
  ${({ size }) => (size ? `font-size: ${size};` : '')}
  ${({ color }) => (color ? `color: ${color};` : '')}
  left: 0;
  margin: 10px;
`;

function Icon({ icon, size, color }: Props) {
  return <IconContainer data-icon={icon} size={size} color={color} />;
}

export default Icon;
