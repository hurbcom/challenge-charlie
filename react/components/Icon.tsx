import { CODES_TO_ICONS, DEVICES } from "@constants/index";
import React from "react";
import styled from "styled-components";

export const WeatherIcon = styled.div<{ icon: string }>`
    font-size: 9rem;
    @media ${DEVICES["tablet"]} {
        font-size: 12rem;
    }
    ::before {
        content: "${({ icon }) => CODES_TO_ICONS[icon]}";
        font-family: "MeteoconsRegular";
    }
`;

export default ({ iconCode }: { iconCode: string }) => {
    return <i  data-icon={CODES_TO_ICONS[iconCode]}></i>;
};
