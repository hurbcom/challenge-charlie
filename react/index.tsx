import React, { StrictMode } from "react";
import { hydrateRoot } from "react-dom/client";
import App from "./App";

const root = document.getElementById("react-app");

if (root) {
    hydrateRoot(
        root,
        <StrictMode>
            <App />
        </StrictMode>
    );
}
