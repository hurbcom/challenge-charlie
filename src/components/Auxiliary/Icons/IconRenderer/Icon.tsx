import React from "react";
import Cloud from "../Cloud";
import CloudMoon from "../CloudMoon";
import CloudSun from "../CloudSun";
import Mist from "../Mist";
import Moon from "../Moon";
import Rain from "../Rain";
import RainMoon from "../RainMoon";
import RainSun from "../RainSun";
import Snow from "../Snow";
import Sun from "../Sun";
import Thunderstorm from "../Thunderstorm";
import TwoClouds from "../TwoClouds";

interface IconRenderProps {
  code: string;
  style: string;
}

interface SelectorInterface {
  [key: string]: () => JSX.Element;
}

/*The icons were made into separate components to avoid clutter on this
selector object and to ease the process of making changes to each one */

const Icon = ({ code, style }: IconRenderProps) => {
  const iconSelector: SelectorInterface = {
    code01d() {
      return <Sun style={style} />;
    },
    code01n() {
      return <Moon style={style} />;
    },
    code02d() {
      return <CloudSun style={style} />;
    },
    code02n() {
      return <CloudMoon style={style} />;
    },
    code03d() {
      return <Cloud style={style} />;
    },
    code03n() {
      return <Cloud style={style} />;
    },
    code04d() {
      return <TwoClouds style={style} />;
    },
    code04n() {
      return <TwoClouds style={style} />;
    },
    code09d() {
      return <Rain style={style} />;
    },
    code09n() {
      return <Rain style={style} />;
    },
    code10d() {
      return <RainSun style={style} />;
    },
    code10n() {
      return <RainMoon style={style} />;
    },
    code11d() {
      return <Thunderstorm style={style} />;
    },
    code11n() {
      return <Thunderstorm style={style} />;
    },
    code13d() {
      return <Snow style={style} />;
    },
    code13n() {
      return <Snow style={style} />;
    },
    code50d() {
      return <Mist style={style} />;
    },
    code50n() {
      return <Mist style={style} />;
    },
  };

  return <>{iconSelector[code]()}</>;
};
export default Icon;
