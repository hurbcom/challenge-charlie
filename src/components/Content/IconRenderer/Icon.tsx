import React from "react";
import Cloud from "../Icons/Cloud";
import CloudMoon from "../Icons/CloudMoon";
import CloudSun from "../Icons/CloudSun";
import Mist from "../Icons/Mist";
import Moon from "../Icons/Moon";
import Rain from "../Icons/Rain";
import Snow from "../Icons/Snow";
import SparseRainMoon from "../Icons/SparseRainMoon";
import SparseRainSun from "../Icons/SparseRainSun";
import Sun from "../Icons/Sun";
import Thunderstorm from "../Icons/Thunderstorm";
import TwoClouds from "../Icons/TwoClouds";

interface IconRenderProps {
  color?: string;
  size?: string;
  code: string;
}

interface SelectorInterface {
  [key: string]: () => JSX.Element;
}

const Icon = ({ color, code, size }: IconRenderProps) => {
  const iconSelector: SelectorInterface = {
    code01d() {
      return <Sun />;
    },
    code01n() {
      return <Moon color={color} size={size} />;
    },
    code02d() {
      return <CloudSun color={color} />;
    },
    code02n() {
      return <CloudMoon color={color} />;
    },
    code03d() {
      return <Cloud color={color} />;
    },
    code03n() {
      return <Cloud color={color} />;
    },
    code04d() {
      return <TwoClouds color={color} />;
    },
    code04n() {
      return <TwoClouds color={color} />;
    },
    code09d() {
      return <Rain color={color} />;
    },
    code09n() {
      return <Rain color={color} />;
    },
    code10d() {
      return <SparseRainSun color={color} />;
    },
    code10n() {
      return <SparseRainMoon color={color} />;
    },
    code11d() {
      return <Thunderstorm color={color} />;
    },
    code11n() {
      return <Thunderstorm color={color} />;
    },
    code13d() {
      return <Snow color={color} />;
    },
    code13n() {
      return <Snow color={color} />;
    },
    code50d() {
      return <Mist color={color} />;
    },
    code50n() {
      return <Mist color={color} />;
    },
  };

  return <>{iconSelector[code]()}</>;
};
export default Icon;
