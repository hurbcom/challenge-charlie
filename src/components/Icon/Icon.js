import React from 'react';

const Icon = (props) => {

  // 'compass' : 'icon-compass',
  // 'clouds-flash' : 'icon-clouds-flash',
  // 'sunrise' : 'icon-sunrise',
  // 'sun' : 'icon-sun',
  // 'moon' : 'icon-moon',
  // 'eclipse' : 'icon-eclipse',
  // 'mist' : 'icon-mist',
  // 'wind' : 'icon-wind',
  // 'snowflake' : 'icon-snowflake',
  // 'cloud-sun' : 'icon-cloud-sun',
  // 'cloud-moon' : 'icon-cloud-moon',
  // 'fog-sun' : 'icon-fog-sun',
  // 'fog-moon' : 'icon-fog-moon',
  // 'fog-cloud' : 'icon-fog-cloud',
  // 'fog' : 'icon-fog',
  // 'cloud' : 'icon-cloud',
  // 'cloud-flash' : 'icon-cloud-flash',
  // 'drizzle' : 'icon-drizzle',
  // 'rain' : 'icon-rain',
  // 'windy' : 'icon-windy',
  // 'windy-rain' : 'icon-windy-rain',
  // 'snow' : 'icon-snow',
  // 'snow-heavy' : 'icon-snow-heavy',
  // 'hail' : 'icon-hail',
  // 'clouds' : 'icon-clouds',
  // 'na' : 'icon-na',
  // 'temperature' : 'icon-temperature',

  return <i className={`icon-${props.name}`} style={props.styles}></i>
}

export default Icon;