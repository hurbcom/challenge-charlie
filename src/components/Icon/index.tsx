import styled from "styled-components";

export type IconSizeType = "small" | "default" | "large" | "xlarge";

const IconBase = styled.i<{ size: IconSizeType }>`
    color: inherit;
    display: flex;
    justify-items: center;
    align-items: center;
    font-style: normal;
    font-size: ${(props) => IconSizesMap[props.size]};

    &:before {
        line-height: 1;
        font-family: "MeteoconsRegular";
        content: attr(data-icon);
    }
`;

const IconNamesMap = {
    morningSun: "A",
    sunny: "B",
    moon: "C",
    eclipse: "D",
    windNigth: "E",
    wind: "F",
    cog: "G",
    cloudDay: "H",
    i: "I",
    j: "J",
    k: "K",
    l: "L",
    m: "M",
    n: "N",
    o: "O",
    p: "P",
    q: "Q",
    r: "R",
    s: "S",
    t: "T",
    u: "U",
    v: "V",
    w: "W",
    x: "X",
    y: "Y",
    z: "Z",
    "0": "0",
    "1": "1",
    "2": "2",
    "3": "3",
    "4": "4",
    "5": "5",
    "6": "6",
    "7": "7",
    "8": "8",
    "9": "9",
    exclamation: "!",
    apostrofe: '"',
    hastatg: "#",
    dollarSign: "$",
    percent: "%",
    andSignal: "&",
    thermometer: "'",
    compass: "(",
    unavailable: ")",
    celsius: "*",
    fahrenheit: "+",
};

const IconSizesMap = {
    small: "12px",
    default: "16px",
    large: "24px",
    xlarge: "32px",
};

type IconPropsType = {
    name: keyof typeof IconNamesMap;
    size?: IconSizeType;
};

const Icon = ({ name, size = "default" }: IconPropsType) => (
    <IconBase role="figure" data-icon={IconNamesMap[name]} size={size} />
);

export default Icon;
