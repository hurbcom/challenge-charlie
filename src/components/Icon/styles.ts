import styled from '@emotion/styled'

import Icon01d from '~/assets/svg/01d.js'
import Icon02d from '~/assets/svg/02d.js'
import Icon03d from '~/assets/svg/03d.js'
import Icon04d from '~/assets/svg/04d.js'
import Icon09d from '~/assets/svg/09d.js'
import Icon10d from '~/assets/svg/10d.js'
import Icon11d from '~/assets/svg/11d.js'
import Icon13d from '~/assets/svg/13d.js'
import Icon50d from '~/assets/svg/50d.js'

export const Svg01d = styled(Icon01d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg02d = styled(Icon02d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg03d = styled(Icon03d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg04d = styled(Icon04d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg09d = styled(Icon09d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg10d = styled(Icon10d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg11d = styled(Icon11d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg13d = styled(Icon13d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`

export const Svg50d = styled(Icon50d)`
  width: 100%;
  height: auto;
  max-width: 27.733rem;

  fill: #ffffff;

  @media (max-width: 767px) {
    max-width: 20.733rem;
  }
`
