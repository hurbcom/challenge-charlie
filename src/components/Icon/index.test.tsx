import { render, screen } from "@testing-library/react";
import Icon, { IconSizeType } from ".";

const IconSizesMap = {
    small: "12px",
    default: "16px",
    large: "24px",
    xlarge: "32px",
};

describe("<Icon /> component", () => {
    test("renders the <Icon/> component with default size", () => {
        const { asFragment } = render(<Icon name="cloudDay" />);

        expect(asFragment).toMatchSnapshot();
        expect(screen.getByRole("figure")).toHaveStyleRule("font-size", "16px");
    });

    test.each([
        ["small", "12px"],
        ["large", "24px"],
        ["xlarge", "32px"],
    ])("renders the <Icon/> component with %s size", (size, expectedValue) => {
        const component = render(
            <Icon name="cloudDay" size={size as IconSizeType} />
        );

        expect(component.asFragment).toMatchSnapshot();
        expect(screen.getByRole("figure")).toHaveStyleRule(
            "font-size",
            expectedValue
        );
    });
});
