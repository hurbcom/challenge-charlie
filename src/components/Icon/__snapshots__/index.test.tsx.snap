// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> component renders the <Icon/> component with default size 1`] = `[Function]`;

exports[`<Icon /> component renders the <Icon/> component with large size 1`] = `[Function]`;

exports[`<Icon /> component renders the <Icon/> component with small size 1`] = `[Function]`;

exports[`<Icon /> component renders the <Icon/> component with xlarge size 1`] = `[Function]`;
